Remotion/Data/DomainObjects.UnitTests/Factories/BaseConfiguration.cs

```diff
@@ -118,5 +124,29 @@ public FakeDomainObjectsConfiguration GetDomainObjectsConfiguration ()
     {
       return new FakeDomainObjectsConfiguration(_mappingLoaderConfiguration, _storageConfiguration, _queryConfiguration);
     }
+
+    public void DisableConnections ()
+    {
+      foreach (var rdbmsProviderDefinition in _storageConfiguration.StorageProviderDefinitions.OfType<RdbmsProviderDefinition>())
+      {
+        var connectionString = rdbmsProviderDefinition.ConnectionString;
+        if (string.IsNullOrEmpty(connectionString))
+          continue;
+
+        _originalConnectionStrings[rdbmsProviderDefinition] = connectionString;
+        PrivateInvoke.SetNonPublicField(rdbmsProviderDefinition, "_connectionString", string.Empty);
```

Comment:
This performance optimization depends on there being just a few switching between database /  no database. What's the execution time difference when caching getting the field instead of using private invoke?

And what's the difference when there is no IF, just the cached field and the set-field-value?

Reply:
The enable/disable code is only run per fixture setup/teardown and takes about 7.6us. With the time being so low I think we can just leave the code as is.